src/ucp/wireup/wireup.c

```diff
@@ -975,6 +967,55 @@ ucp_wireup_get_reachable_mds(ucp_worker_h worker,
     key->reachable_md_map = dst_md_map;
 }
 
+static void
+ucp_wireup_check_config_intersect(ucp_ep_h ep, ucp_ep_config_key_t *new_key)
+{
+    ucp_worker_h worker = ep->worker;
+    uct_ep_h new_uct_eps[UCP_MAX_LANES] = {NULL};
+    ucp_lane_index_t reuse_lane_map[UCP_MAX_LANES] = { UCP_NULL_LANE };
+    ucp_ep_config_key_t *old_key;
+    ucp_lane_index_t lane, reuse_lane;
+
+    if (ep->cfg_index == 0) {
+        return;
+    }
+
+    old_key = &ucp_ep_config(ep)->key;
+
+    ucp_ep_config_lanes_intersect(old_key, new_key, reuse_lane_map);
+
+    for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
+        ucs_assert(ep->uct_eps[lane] != NULL);
+    }
+
+    /* CM lane has to be re-used by the new EP configuration */
+    ucs_assert(reuse_lane_map[ucp_ep_get_cm_lane(ep)] != UCP_NULL_LANE);
+
+    /* Need to discard only old lanes that won't be used anymore in the new
+     * configuration. Also, UCT EPs with the lane index >= old_key->num_lanes
+     * could be set in case of CM, we have to not reset them */
+    for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
+        reuse_lane = reuse_lane_map[lane];
+        if (reuse_lane == UCP_NULL_RESOURCE) {
+            if (ep->uct_eps[lane] != NULL) {
+                // TODO: discard unused uct_ep
+                ucs_warn("!!!worker need reclaim resource");
+                (void)(worker);
```

Comment:
try ```uct_ep_destroy``` or ```close``` & **uet_ep_pending_purge**
port **ucp_request_purge_enqueue_cb** to int3 (purge all reqs on pending queue)